lesson-04-memory-management/mm/xxx.c

```diff
@@ -5,21 +5,45 @@
 #include <linux/pci.h>
 #include <linux/version.h>
 #include <linux/init.h>
+#include <linux/timer.h>
 
-#define LEN_MSG 160
-static char buf_msg[ LEN_MSG + 1 ] = "Hello from module!\n";
+static int n1, n2;
+static int n[1];
+static void fib_handler( unsigned long );
+DEFINE_TIMER( mytimer, fib_handler, 0, 0 );
 
+void fib_handler( unsigned long data )
+{
+  if (*n)
+  {
+    int t = n2;
+    n2 = n1 + n2;  
+    if (n2 < 0)
+    {
+      n1 = 0; n2 = 1;
+    }
+    else  
+      n1 = t;
+
+    *n = n2; //atomic
+  }
+  else
+  {
+    n1 = 0; n2 = 1;
```

Comment:
Looks like `*n` won't ever be updated.
PRs should be tested somehow before publishing.